docs/introduction/docker-stream-logging.md

```diff
@@ -0,0 +1,104 @@
+```
+For other docker containers we can easily setup logging of any service inside container like it is for php-fpm.
+
+## Monolog
+Php application uses [symfony/monolog-bundle](https://github.com/symfony/monolog-bundle) for logging. It is easy to setup
+monolog to send logs into streams by modifiying `app/config/config.yml`
+```yaml
+monolog:
+  channels: ["cron", "slow"]
+  handlers:
+    cron:
+      type: stream
+      path: "php://stdout"
+      channels: cron
+      level: debug
+    slow:
+      type: stream
+      path: "php://stdout"
+      channels: slow
+      level: debug
+```
+
+## Docker logging drivers
+Each container transfers all received stream logs with its assigned [logging driver](https://docs.docker.com/config/containers/logging/configure/#supported-logging-drivers).
+Logging driver can be set for docker container in `docker-compose.yml`
+```yaml
+version: "3.1"
+services:
+    postgres:
+        image: postgres:9.5-alpine
+        container_name: shopsys-framework-postgres
+        logging:
+          driver: syslog
+          options:
+            syslog-address: udp://host_with_syslog_daemon:514
+```
+
+### Json-file driver
+Default setting of logging driver for each docker container is [json-file](https://docs.docker.com/config/containers/logging/json-file/)
+This driver stores logs as json files in `/var/lib/docker/containers/container_id/` thus the log files are divided
+can be rotated by size and limited by files count.
+
+### Syslog driver
+To have centralized logs from multiple containers is better to use syslog driver that send logs in its specific form to
+server or daemon that can handle syslog messages.
+
+Syslog daemon can be established for instance on localhost
+```yaml
+php-fpm:
+    logging:
+        driver: syslog
+        options:
+            syslog-address: udp://localhost:514
+	tag: docker
+```
+
+#### Rsyslog setup
+```
+sudo apt install rsyslog
+```
+in file `/etc/rsyslog.conf` we need to allow listening on port
+```
+module(load="imudp")
+input(type="imudp" port="514")
+```
+```sh
+sudo service rsyslog restart 
+```
+as a default, syslog stores all incoming messages into `/var/log/syslog` but there is a way how to filter logs that come
+from docker containers and store them into different path by modifying rsyslog configuration file so before
+line `*.*;auth,authpriv.none -/var/log/syslog` we can put
+```
+if $syslogtag startswith "docker" and $msg contains "cron." then /path_to_logs/docker-php-fpm.cron.log
+& stop # stop processing of matched message 
+if $syslogtag startswith "docker" and $msg contains "slow." then /path_to_logs/docker-php-fpm.slow.log
+& stop
+if $syslogtag startswith "docker" then /path_to_logs/logs.log
+& stop
+```
+
+## Conclusion
+
+Since we need another container for the centralized logging system and additional setup, we will postpone it until 
+implementation of orchestration system like Kubernetes that will handle all streams from docker containers unified way.
```

Comment:
The last acceptance criterium is not done. I'm not sure why is not possible to configure new container for logs even before orchestration. I was expecting that after this US there will be logging problem solved.